snapcraft/storeapi/_snap_index_client.py

```diff
@@ -46,40 +63,37 @@ def get_default_headers(self):
 
         return headers
 
-    def get_package(self, snap_name, channel=None, arch=None):
-        """Get the details for the specified snap.
+    def get_info(self, snap_name: str, *, arch: str = None) -> SnapInfo:
+        """Get the information for the specified snap.
 
         :param str snap_name: Name of the snap.
-        :param str channel: Channel of the snap.
         :param str arch: Architecture of the snap (none by default).
 
-        :return Details for the snap.
-        :rtype: dict
+        :return information for the snap.
+        :rtype: SnapInfo
         """
         headers = self.get_default_headers()
         headers.update(
             {
                 "Accept": "application/hal+json",
-                "X-Ubuntu-Series": constants.DEFAULT_SERIES,
+                "Snap-Device-Series": constants.DEFAULT_SERIES,
             }
         )
-        if arch:
-            headers["X-Ubuntu-Architecture"] = arch
-
-        params = {
-            # FIXME LP: #1662665
-            "fields": "status,confinement,anon_download_url,download_url,"
-            "download_sha3_384,download_sha512,snap_id,"
-            "revision,release"
-        }
-        if channel is not None:
-            params["channel"] = channel
-        logger.debug("Getting details for {}".format(snap_name))
-        url = "api/v1/snaps/details/{}".format(snap_name)
+
+        params = dict()
+        params[
+            "fields"
+        ] = "channel-map, snap,snap-id,name,publisher,confinement,revision"
```

Comment:
There is a stray space before `snap`, but this field doesn't not exist and it's not needed anyway.

Reply:
removed, valid fields are not documented though, are they? they also seem to be flattened, correct?